### PR TITLE
refactor: Port sync with cards

### PR DESCRIPTION
**Pull request by kartikCypherock**
Implements https://app.clickup.com/t/9002019994/PRF-4860

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4860 Sync with cards](https://app.clickup.com/t/9002019994/PRF-4860)

File: src/settings/sync_with_cards.c [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+    memcpy(new_wallet.wallet_name, wallet_list->wallet[index].name, NAME_SIZE);
+    memcpy(new_wallet.wallet_id, wallet_list->wallet[index].id, WALLET_ID_SIZE);
+
+    // Record card number on which the wallet is locked
+    if (new_wallet.is_wallet_locked) {
+      new_wallet.challenge.card_locked = *tapped_card;
+    }
+
+    uint32_t dummy;
+    add_wallet_to_flash(&new_wallet, &dummy);
```
Comment (amanCypherock) [inline src/settings/sync_with_cards.c:176]: Please check the return code here. Either assert or return the error code.

File: src/settings/sync_with_cards.c [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+
+    snprintf(msg,
+             sizeof(msg),
+             UI_TEXT_SYNC_WALLET_PROMPT,
+             (char *)wallet.wallet_name);
+    if (!core_scroll_page(NULL, msg, NULL)) {
+      continue;
+    }
+
+    sync_state_e flow_state = sync_wallets_flow();
```
[the remark this answers is not in the document]
Comment (amanCypherock) [reply src/settings/sync_with_cards.c:266]: Particularly the wallet structure population, and the confirmation screen.

[the remark this answers is not in the document]
Comment (kartikCypherock) [reply src/settings/sync_with_cards.c:266]: As per https://github.com/Cypherock/x1_wallet_firmware/pull/358, a confirmation is already there on the wallet menu (see case `VALID_WALLET_WITHOUT_DEVICE_SHARE`)

So moving confirmation in the flow doesn't make sense..

Wallet structure population, I will move.. Maybe pass `wallet_id` as the argument..

File: src/settings/sync_with_cards.c (hunk 1 above)
Comment (kartikCypherock) [reply src/settings/sync_with_cards.c:176]: Ok, sure

Comment (kartikCypherock) [reply src/settings/sync_with_cards.c]: Added the handling

File: src/settings/sync_with_cards.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (kartikCypherock) [reply src/settings/sync_with_cards.c]: Updated

**Review by amanCypherock — APPROVED** [review]

**Pull request merged**